mantidproject/newstarter
# Brandon Hewer: Ex01 and Ex02

## Exercise 1
A unique word counter for ASCII-formatted files. Counts the number of occurrences of each unique word (of length greater than 4). Case and punctuation insensitive; punctuation is considered to be the set of characters ".,?'"!():"

**To test:**
  1. Build the WordCounter project.
  2. a) Run the build, passing the directory of the ASCII file to count from as an argument.
      b) To check error handling, pass in an invalid directory.
      c) To check error handling, do not pass an argument when running the build.
  3. Once application has quit, open the 'results.txt' file in the build directory.
  4. Check to see that the 'results.txt' file contains a list of unique words and their total number of
      occurrences each on a new line.

## Exercise 2
An application for creating, sorting and filtering 2-dimmensional shapes. Shapes included: squares, rectangles, circles, triangles. Provides sorting by perimeter and area, as well as filtering by type (name) and number of sides.

**To test:**
  1. Build the Shapes project.
  2. a) Construct one of each shape (square, rectangle, circle and triangle), by calling their respective
          constructors with positive double values as arguments (one for circle and square, two for
          rectangle and triangle).
      b) To test error handling, try constructing the shapes by passing negative double values as arguments
           to their constructors (Note: A Runtime Exception should be thrown).
  3. Create a vector of pointers to the constructed Shapes.
  4. Construct a ShapeSorter by calling its constructor and passing in the previously created vector as
      an argument.
  5. a) Print out the returned values upon calling the filter methods, filterByType and filterBySideNumber 
          when passing in each type ('Circle', 'Square', 'Triangle, 'Rectangle') and side number (1, 3, 4) 
          respectively.
      b) Print out the returned values upong calling the same methods with types or side numbers which
          do not correspond to any shapes (should not display any shapes).
  6. Print out the returned values upon calling the sort methods, sortByPerimeter and sortByArea, when
      passing in both directional values ShapeSorter::ASCENDING (true) and ShapeSorter::DESCENDING 
      (false).
